compaction/compaction_manager.cc

```diff
@@ -1132,18 +1151,24 @@ class compaction_manager::rewrite_sstables_compaction_task : public compaction_m
     {}
 
 protected:
-    virtual future<> do_run() override {
+    virtual future<compaction_result_counters> do_run() override {
+        sstables::compaction_counters counters{};
+
         switch_state(state::pending);
         auto maintenance_permit = co_await acquire_semaphore(_cm._maintenance_ops_sem);
 
         while (!_sstables.empty() && can_proceed()) {
             auto sst = consume_sstable();
-            co_await rewrite_sstable(std::move(sst));
+            const auto& res = co_await rewrite_sstable(std::move(sst));
+            counters += res.counters;
         }
+        _cm.validation_errors += counters.validation_errors;  // TODO Deexie  maybe it should be done after every rewrite_sstable to provide gradual growth (?)
```

Comment:
It would be better to update `_cm.validation_errors` in the inner loop, per sstable as the comment says.